email/add_ecr/email_sender.py

```diff
@@ -0,0 +1,311 @@
+        password=os.getenv('DB_PASSWORD'),
+        port=os.getenv('DB_PORT')
+    )
+
+
+def get_publishers_tracked_authors(publisher_id: int) -> list[tuple]:
+    '''Gets all publishers  and their tracked authors from the DB'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            a.author_id
+            FROM publisher as p
+            LEFT JOIN author_assignment AS aa ON aa.publisher_id = p.publisher_id
+            LEFT JOIN author AS a ON a.author_id = aa.author_id
+            where p.publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchall()
+    finally:
+        conn.close()
+
+
+def get_publisher_ids() -> list[tuple]:
+    '''Gets the ids of all publishers'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_id
+            FROM publisher
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchall()
+    finally:
+        conn.close()
+
+
+def get_publishers_name(publisher_id: int) -> str:
+    '''Returns the name for the specified publisher id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_name
+            FROM publisher
+            WHERE publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchone()[0]
+    finally:
+        conn.close()
+
+
+def get_publishers_email(publisher_id: int) -> str:
+    '''Returns the email for the specified publisher id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_email
+            FROM publisher
+            WHERE publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchone()[0]
+    finally:
+        conn.close()
+
+
+def get_avg_rating_difference_since_yesterday(author_id) -> int:
+    '''gets ratings since yesterday for a given author_id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            average_rating - LAG(average_rating) OVER (
+                ORDER BY am.date_recorded DESC) avg_change_since_yesterday
+            FROM author AS a
+            JOIN author_measurement AS am ON a.author_id = am.author_id
+            WHERE am.author_id = {author_id}
+            ORDER BY am.date_recorded DESC
+            LIMIT 2
+            ;
+            '''
+        cur.execute(sql)
+        rating_change = cur.fetchall()
+        if not rating_change:
+            return 'No historical data for this author yet.'
+        return rating_change[1][0]
+    finally:
+        conn.close()
+
+
+def get_shelved_difference_from_yesterday(author_id) -> int:
```

Comment:
Another missing type hint here for `author_id` 
```suggestion
def get_shelved_difference_from_yesterday(author_id: int) -> int:
```